src/Database/Database.php

```diff
@@ -134,8 +135,12 @@ public function connect()
 			if ($charset) {
 				$connect .= ";charset=" . $charset;
 			}
-
-			try {
+			
+                        if ($socket) {
+                                $connect .= ";$unix_socket=" . $socket;
```

Comment:
could the unix_socket param be in conflict with a port parameter? Should we catch it beforehand?

Reply:
As far as I'm aware they shouldn't conflict? The type of connection is determined by the value of the hostname parameter (more info on that [here](https://www.php.net/manual/en/mysqli.construct.php)). If it's localhost or null, sockets/pipes will be used. It looks like you could provide both parameters, and if you pass an IP address (or localhost / null) to the hostname parameter, the socket (or port) parameter should be ignored.